tests/pipeline/nodes/base/test_threshold_checker_mixin.py

```diff
@@ -29,419 +29,418 @@ def __init__(self):
+            threshold_model.check_bounds({"key1": "a"}, "[10, +inf]")
+        assert "`key` must be either str or list" == str(excinfo.value)
+
+    def test_check_bounds_bad_format(self, threshold_model):
+        with pytest.raises(ValueError) as excinfo:
+            # missing separator
+            threshold_model.check_bounds("a", "[13 14]")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # wrong separator
+            threshold_model.check_bounds("a", "[13; 14]")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # missing left bracket
+            threshold_model.check_bounds("a", "13, 14]")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # missing right bracket
+            threshold_model.check_bounds("a", "[13, 14")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # wrong left bracket
+            threshold_model.check_bounds("a", "{13, 14]")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # wrong right bracket
+            threshold_model.check_bounds("a", "[13, 14}")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # wrong right bracket
+            threshold_model.check_bounds("a", "[13, 14}")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # wrong left value
+            threshold_model.check_bounds("a", "[value, 14]")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # wrong left value
+            threshold_model.check_bounds("a", "[1 2, 14]")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            # wrong right value
+            threshold_model.check_bounds("a", "[13, 14.5.6]")
+        assert "Badly formatted interval" == str(excinfo.value)
+
+    def test_check_bounds_invalid_interval(self, threshold_model):
+        with pytest.raises(ValueError) as excinfo:
+            threshold_model.check_bounds("a", "[14, 13]")
+        assert "Lower bound cannot be larger than upper bound" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            threshold_model.check_bounds("a", "[+inf, 13]")
+        assert "Lower bound cannot be larger than upper bound" == str(excinfo.value)
+
+        with pytest.raises(ValueError) as excinfo:
+            threshold_model.check_bounds("a", "[14, -inf]")
+        assert "Lower bound cannot be larger than upper bound" == str(excinfo.value)
```

Comment:
Maybe add the test `[+inf, -inf]`?
Is `[0, 0]` "allowed"/"valid"?

Reply:
Added test case for `[+inf, -inf]`.

`[0, 0]` is a valid interval, it will be the equivalent of doing `x == 0.0`. Surprisingly `(0, 0)` is a valid interval as well, the comparison will always fail as there is not valid values